contracts/MarginBase.sol

```diff
@@ -412,4 +454,148 @@ contract MarginBase is MinimalProxyable {
     {
         return IFuturesMarket(futuresManager.marketForKey(_marketKey));
     }
+
+    /// @notice exchangeRates() fetches current ExchangeRates contract
+    function exchangeRates() internal view returns (IExchangeRates) {
+        return
+            IExchangeRates(
+                addressResolver.requireAndGetAddress(
+                    "ExchangeRates",
+                    "MarginBase: Could not get ExchangeRates"
+                )
+            );
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            Limit Orders
+    ///////////////////////////////////////////////////////////////*/
+
+    /// @notice limit order logic condition checker
+    /// @param _orderId: key for an active order
+    function validOrder(uint256 _orderId) public view returns (bool) {
+        Order memory order = orders[_orderId];
+
+        bytes32 currencyKey = futuresMarket(order.marketKey).baseAsset();
+        // Get exchange rate for 1 unit
+        uint256 price = exchangeRates().effectiveValue(
+            currencyKey,
+            1e18,
+            "sUSD"
```

Comment:
Up to you, but we have TRACKING_CODE as a constant, may make this a constant too